### ISSUE TITLE
Option to use external version of node

### ISSUE THREAD

**Issue by tehbilly** [s-a/iron-node#62]
As much as I'm a fan of electron, and I am, using that runtime to debug scripts or applications written to run outside of electron will lead to an imperfect test environment.

Wondering what the feasibility is of using the standard remote debugging flags and utilities alongside `exec`-ing the entry file via the standard node executable. I like the interface and capabilities of this application, and decoupling the electron runtime from the environment I'm debugging would allow for things like testing a problem under different versions. Seems like it would also solve some of the other open issues.

**Comment by s-a** [comment]
Sorry the node version is compiled within iron-node eg electron. So afaik it is not pssible to target another version of node.js.

**Comment by tehbilly** [comment]
Not without spawning another process, which would also allow you to do cool things like mentioned in #64. I took a glance and couldn't find where you're using `vm` or how you're executing the file you pass as an argument, but essentially you'd be switching to something like:

``` javascript
let spawn = require('child_process').spawn;
let theSpawned = spawn('node', ['--debug-brk', pathToScript.js]);
```

Then connect to the debugger over TCP. Making iron-node a general-purpose debugger with a default of a local file being the target makes a bit more sense to me. Electron is not node, and assuming a version of node being available outside of the electron runtime is pretty safe since the readme suggests installing iron-node via npm.

If you can point me at where execution of the script is done I can take a stab at it, but I'm at work and can't hunt but so much on my own right now.

**Comment by s-a** [comment]
This sounds interesting. You should talk to @mark-hahn about your idea. 
The entry point for the debug session is at 
https://github.com/s-a/iron-node/blob/master/app/app.js#L191
We should recode the if block at  https://github.com/s-a/iron-node/blob/master/app/app.js#L186
For the remote debug session I could imagine a menu point from GUI to target the remote URL but How to tell iron-node to use local installed Node Version ? Maybe we could configure this in the local or global configuration file. Then it could be possible to point to specific installed Node version without Parameter pollution.

**Comment by mark-hahn** [comment]
I started an Atom debugger package a while back that talked to a separate process running the chrome debugger feature.  I got everything working including breakpoints, etc. but when I ran into the work to display variables I burned out.  This  worked for any browser JS and also for other atom instances.  My motivation was to debug atom packages.

So I have existing code that talks to the chrome debugger.  I'm hoping to merge it with this project when I get a chance.
